addons/web_editor/controllers/main.py

```diff
@@ -607,6 +610,32 @@ def modify_image(self, attachment, res_model=None, res_id=None, res_field=None,
         :param saved_image_attributes: The options of the modified image
         :param res_type: The type of the closest editable element to the image
         """
+        if not attachment:
+            # Create an original attachment first.
+            mimetype_original = saved_image_attributes['mimetype_before_conversion']
+            name_original = '%s-%s%s' % (
+                datetime.now().strftime('%Y%m%d%H%M%S'),
+                str(uuid.uuid4())[:6],
+                SUPPORTED_IMAGE_MIMETYPES[mimetype_original],
+            )
+            if res_type != 'image':
+                # HTML image
+                attachment = request.env['ir.attachment'].create({
```

Comment:
Note that something like
```py
req = requests.get(url)
attachment = request.env['ir.attachment'].create({
                'name': name_original,
                'mimetype': req.headers['content-type'],
                'public': True,
                'raw': req.content,
                'res_model': 'ir.ui.view',
                'res_id': 0,
            })
```
where the image is first fetched could also have been done to follow the logic of [this commit](https://github.com/odoo/odoo/commit/943944dd249c15de870d6800d89e48d54a422e5a).
However, I assumed that an image that would enter this code is an internal image used by default in a view. I so decided to rather create an attachment of type 'url' to follow the logic used for the default image snippet for example.